Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ComponentEditor.cpp

```diff
@@ -930,18 +930,18 @@ namespace AzToolsFramework
             componentModeTypes.end(), m_componentType) == componentModeTypes.end())
         {
             SetWidgetInteractEnabled(this, false);
-            return ComponentEditor::ComponentEditorState::DidNotEnterComponentMode;
+            return ComponentEditor::ComponentEditorState::ComponentCardDisabled;
         }
         else
         {
             if (!componentModeTypes.empty())
             {
                 // only set the first item to be selected/highlighted
                 SetSelected(componentModeTypes.front() == m_componentType);
-                return ComponentEditor::ComponentEditorState::DidEnterComponentMode;
+                return ComponentEditor::ComponentEditorState::ComponentCardSelected;
             }
         }
-        return ComponentEditor::ComponentEditorState::DidNotEnterComponentMode;
+        return ComponentEditor::ComponentEditorState::ComponentCardDisabled;
```

Comment:
update: Sorry I think we actually need three enum states here as if we get to this point, the component card hasn't been updated, it's been left 'as-is' (not disabled and not selected). Try running the code path when you have a Spline and Tube Shape on an entity at the same time to see what I mean. Can you please add this case too and return it here 'ComponentCardUnmodified' maybe?

Reply:
I am pretty certain the final state can be reached, but it only happens in the case of something like the Tube Shape Component and Spline Component both being on the Entity at the same time and Tube Shape Component Mode starts. We keep both Spline and Tube active, but one is the 'selected' one (for the Component Mode you're in). That's what I'm saying the enum should reflect

Reply:
Tom and I looked into this further and it could not be reached but we want it to be reached to accurately reflect what's happening with the cards. We were returning that both cards were selected when the Tube's component mode was active but only the Tube card was being selected while the Spline card was being left alone (but returning that it was selected) The change I just pushed should make it more clear